### DISCUSSION TITLE
How can I use app dir to perform the same behavior as getServerSideProps?

### DISCUSSION THREAD

**Discussion by es-ki** [vercel/next.js#48281] [Help] [answered]
### Summary

I read on [this page](https://beta.nextjs.org/docs/api-reference/fetch) that adding the ```{ cache: 'no-store' }``` option to fetch would result in the same behavior as getServerSideProps, but when deployed, SSR did not work.
On the localhost, fetch was running on every request.
Here is the code, am I missing something to do SSR?
```tsx
export default async function Search({ searchParams }) {
  const query = searchParams.q;
  const res = await fetch(
    `${process.env.NEXT_PUBLIC_APP_URL}/search/api?q=${query}`,
    {
      cache: "no-store",
    }
  );
  const { data } = await res.json();

  return (
    <div>
      {data.length > 0 && (
        <ul>
          {data.map((d) => {
            const { id, title } = d;
            return <li key={id}>{title}</li>;
          })}
        </ul>
      )}
    </div>
  );
}
```

### Additional information

```js
The version of Next.js I am using is 13.3.1-canary.4.
```

**Comment by icyJoseph** [comment]
I think that this is a problem with the beta stuff... You could use the `force-dynamic` segment config instead:

https://beta.nextjs.org/docs/api-reference/segment-config#dynamic

But it feels like we miss out on the granularity of having the fetch requests we want, to be SSR'd

**Reply by es-ki** [reply]
Thank you.
I tried using the force-dynamic option but could not improve it.
I was using the cms client library within Route Handlers to retrieve data, but it seemed that the fetch in the client library was cached.

Developers using cms and the cms client library may face this problem.

**Comment by es-ki** [comment, marked as answer]
This was due to the cms client library used within Route Handlers.
It seems that the fetch contained in the client library was cached.
By executing the restful api directly without using the client library, ```cache: "no-store"``` worked.
